src/layers/CasaLayer/CasaLayer.js

```diff
@@ -0,0 +1,172 @@
+ * @param {Object} options Layer options.
+ * @param {String} options.apiKey Access key for [geOps services](https://developer.geops.io/).
+ * @param {styleFunction} [options.styleFunction] Style function.
+ */
+class CasaLayer extends VectorLayer {
+  constructor(options = {}) {
+    super(options);
+
+    this.apiKey = options.apiKey;
+
+    this.projection = options.projection || 'EPSG:3857';
+
+    this.defaultStyleObject = {
+      stroke: {
+        color: 'rgb(255, 200, 25)',
+        width: 2,
+      },
+      strokeOutline: {
+        color: 'white',
+        width: 8,
+      },
+      fill: {
+        color: 'rgb(255, 200, 25)',
+      },
+      text: {
+        font: '12px Arial',
+        color: 'black',
+        label: 'zone',
+      },
+      textOutline: {
+        width: 2,
+        color: 'white',
+      },
+    };
+
+    this.styleFunction = options.styleFunction || this.defaultStyleFunction;
+  }
+
+  defaultStyleFunction(properties, isSelected) {
+    return isSelected
+      ? {
+          ...this.defaultStyleObject,
+          stroke: {
+            color: 'blue',
+            width: 4,
+          },
+        }
+      : this.defaultStyleObject;
+  }
+
+  /**
+   * Converts a zone style to an ol.Style.
+   * @private
+   * @param {styleObject} [styleObject = {}] Style object.
+   * @param {number} resolution Map resultion.
+   * @param {boolean} [isSelected = false] Whether the feature is selected.
+   */
+  getOlStyleFromObject(styleObject = {}, isSelected = false) {
+    const style = deepmerge(this.defaultStyleObject, styleObject);
+
+    const olStyle = [
+      new Style({
+        stroke: new StrokeStyle({
+          ...style.stroke,
+        }),
+        fill: new FillStyle({
+          ...style.fill,
+        }),
+        text: new TextStyle({
+          font: style.text.font,
+          fill: new FillStyle({
+            color: style.text.color,
+          }),
+          stroke: new StrokeStyle({
+            ...style.textOutline,
+          }),
+          text: style.text.label,
+        }),
+      }),
+    ];
+
+    if (style.strokeOutline) {
+      olStyle.unshift(
+        new Style({
+          stroke: new StrokeStyle({
+            ...style.strokeOutline,
+          }),
+        }),
+      );
+    }
+
+    if (isSelected) {
+      olStyle.forEach(s => s.setZIndex(1));
+    }
+
+    return olStyle;
+  }
+
+  /**
+   * Override the callClickCallbacks function of react-spatial's Vectorlayer
+   * in order to stop the propagation if the first feature is from another layer.
+   * @private
+   * @inheritdoc
+   */
+  callClickCallbacks(features, layer, coordinate) {
+    const pixel = this.map.getPixelFromCoordinate(coordinate);
+    const featuresAtPixel = [];
+
+    this.map.forEachFeatureAtPixel(pixel, f => {
+      featuresAtPixel.push(f);
+    });
+
+    const isTopFeature = this.olLayer
+      .getSource()
+      .getFeatures()
+      .includes(featuresAtPixel.slice(0, 1)[0]);
+
+    if (!isTopFeature) {
```

Comment:
You could use `getFeaturesAtPixel`  with a` layerFIlter`, it will simplify:
https://openlayers.org/en/latest/apidoc/module-ol_Map-Map.html#getFeaturesAtPixel